pkg/diskmaker/controllers/lvset/matcher.go

```diff
@@ -0,0 +1,167 @@
+package lvset
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	localv1alpha1 "github.com/openshift/local-storage-operator/pkg/apis/local/v1alpha1"
+	"github.com/openshift/local-storage-operator/pkg/internal"
+	"golang.org/x/sys/unix"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+const (
+	// filter names:
+	notReadOnly           = "notReadOnly"
+	notRemovable          = "notRemovable"
+	notSuspended          = "notSuspended"
+	noFilesystemSignature = "noFilesystemSignature"
+	// file access , can't mock test
+	noChildren = "noChildren"
+	// file access , can't mock test
+	canOpenExclusively = "canOpenExclusively"
+
+	// matchers names:
+	inSizeRange              = "inSizeRange"
+	inTypeList               = "inTypeList"
+	inMechanicalPropertyList = "inMechanicalPropertyList"
+	inVendorList             = "inVendorList"
+	inModelList              = "inModelList"
+)
+
+// maps of function identifier (for logs) to filter function.
+// These are passed the localv1alpha1.DeviceInclusionSpec to make testing easier,
+// but they aren't expected to use it
+// they verify that the device itself is good to use
+var filterMap = map[string]func(internal.BlockDevice, localv1alpha1.DeviceInclusionSpec) (bool, error){
+	notReadOnly: func(dev internal.BlockDevice, spec localv1alpha1.DeviceInclusionSpec) (bool, error) {
```

Comment:
I didn't have it initially, but keeping it generic makes it much easier to unit test.

Reply:
I do not understand why will a function receive an argument that it never uses. How does it make unit testing easier? You can just choose to not supply that argument..

Reply:
In this case: https://github.com/rohantmp/local-storage-operator/blob/8b2aa777ac8416bdd25ba103e50fcb8947430737/pkg/diskmaker/controllers/lvset/matcher_test.go#L540

Reply:
But yeah, it was mostly a refactoring concern and I can remove it.

Reply:
Fixed. It now accepts a pointer to the spec, but I kept it with the same signature as matcherFunc so that they could be unit tested with the same funcs